core/files/configure_misp.sh

```diff
@@ -72,46 +72,72 @@ GPGEOF
+            \"OidcAuth\": {
+                \"provider_url\": \"${OIDC_PROVIDER_URL}\",
+                ${OIDC_ISSUER:+\"issuer\": \"${OIDC_ISSUER}\",}
+                \"client_id\": \"${OIDC_CLIENT_ID}\",
+                \"client_secret\": \"${OIDC_CLIENT_SECRET}\",
+                \"roles_property\": \"${OIDC_ROLES_PROPERTY}\",
+                \"role_mapper\": ${OIDC_ROLES_MAPPING},
+                \"default_org\": \"${OIDC_DEFAULT_ORG}\"
+            }
+        }" > /dev/null
 
-    sudo -u www-data php /var/www/MISP/tests/modify_config.php modify "{
-        \"OidcAuth\": {
-            \"provider_url\": \"${OIDC_PROVIDER_URL}\",
-            ${OIDC_ISSUER:+\"issuer\": \"${OIDC_ISSUER}\",}
-            \"client_id\": \"${OIDC_CLIENT_ID}\",
-            \"client_secret\": \"${OIDC_CLIENT_SECRET}\",
-            \"roles_property\": \"${OIDC_ROLES_PROPERTY}\",
-            \"role_mapper\": ${OIDC_ROLES_MAPPING},
-            \"default_org\": \"${OIDC_DEFAULT_ORG}\"
-        }
-    }" > /dev/null
+        # Set the custom logout URL for OIDC if it is defined
+        if [[ -n "${OIDC_LOGOUT_URL}" ]]; then
+            sudo -u www-data /var/www/MISP/app/Console/cake Admin setSetting -q "Plugin.CustomAuth_custom_logout" "${OIDC_LOGOUT_URL}&post_logout_redirect_uri=${BASE_URL}/users/login"
+        else
+            echo "OIDC_LOGOUT_URL is not set"
+        fi
+
+        # Disable password confirmation as recommended in https://github.com/MISP/MISP/issues/8116
+        sudo -u www-data /var/www/MISP/app/Console/cake Admin setSetting -q "Security.require_password_confirmation" false
+
+        echo "... OIDC authentication enabled"
 
-    # Set the custom logout URL for the OIDC plugin only if OIDC_LOGOUT_URL is defined
-    if [[ -n "${OIDC_LOGOUT_URL}" ]]; then
-        sudo -u www-data /var/www/MISP/app/Console/cake Admin setSetting -q "Plugin.CustomAuth_custom_logout" "${OIDC_LOGOUT_URL}&post_logout_redirect_uri=${BASE_URL}/users/login"
     else
-        echo "OIDC_LOGOUT_URL is not set"
-    fi
+        # Reset OIDC authentication settings to empty values
+        sudo -u www-data php /var/www/MISP/tests/modify_config.php modify "{
+            \"OidcAuth\": {
+                \"provider_url\": \"\",
+                \"issuer\": \"\",
+                \"client_id\": \"\",
+                \"client_secret\": \"\",
+                \"roles_property\": \"\",
+                \"role_mapper\": \"\",
+                \"default_org\": \"\"
+            }
+        }" > /dev/null
 
-    # Disable password confirmation as stated at https://github.com/MISP/MISP/issues/8116
-    sudo -u www-data /var/www/MISP/app/Console/cake Admin setSetting -q "Security.require_password_confirmation" false
+        # Use sed to remove the OidcAuth.Oidc entry from the 'auth' array in the config.php
+        sudo sed -i "/'auth' =>/,/)/ { /0 => 'OidcAuth.Oidc',/d; }" /var/www/MISP/app/Config/config.php
```

Comment:
Use `sudo -u www-data` here

Reply:
like this `sudo -u www-data sed -i "/'auth' =>/,/)/ { /0 => 'OidcAuth.Oidc',/d; }" /var/www/MISP/app/Config/config.php`

Reply:
Yes, all commands that edit MISP state and file should use the same user/permission used by MISP